Remove embargo date when object is not embargoed

hortongn opened the pull request:
The change in hydra-access-controls that was needed for this fix has been merged into hydra-head.  So I'm opening a PR for the related changes in Curate.  This will allow an object with an embargo date to be switched to different access rights with no release date.

hortongn committed:
Remove embargo date when object is not embargoed

hortongn committed:
Remove embargo date when object is not embargoed

jeremyf commented:
@hortongn Can you rebase this commit? I minted a new Hydra Access Control gem (6.4.2 if memory serves)

hortongn committed:
Merge branch 'feature/HYDRASIR-296' of github.com:projecthydra-labs/curate into feature/HYDRASIR-296

hortongn commented:
I'll take a look and see why tests are failing

hortongn committed:
Better fix for moving works from embargo to public

Undo previous changes to app/repository_models/curation_concern/embargoable.rb
and implement a simpler and better fix for users not being able to
move a work from embargoed status to open/public.  The problem is
that the setter method for embargo_release_date= gets called twice
and on the second call it adds the embargo date again to
@embargo_release_date.  So I added a conditional that only sets
@embargo_release_date once (to nil) if the user is trying to take
something out of embargo.

Fixing embargoable.rb caused some of the embargo tests to break even
though the app was performing as it should.  So I tweaked the tests
accordingly.
